packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -186,7 +189,7 @@ export const useGridSelection = (apiRef: GridApiRef, props: GridComponentProps):
   React.useEffect(() => {
     // Rows changed
     setGridState((state) => {
-      const newSelectionState = [...state.selection];
+      const newSelectionState = getArraySelectionModel(state.selection);
```

Comment:
Reuse `gridArraySelectionStateSelector`

```suggestion
      const newSelectionState = gridArraySelectionStateSelector(state);
```

Same for the `isRowSelectable`'s effect.